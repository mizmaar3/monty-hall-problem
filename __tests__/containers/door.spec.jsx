import React from "react";
import { shallow } from "enzyme";
import Door from "containers/door";

describe("Door container", () => {
  it("Should render door correctly with default props", () => {
    const wrapper = shallow(<Door />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render door correctly with open prop", () => {
    const wrapper = shallow(<Door open={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render door correctly with selected prop", () => {
    const wrapper = shallow(<Door open={true} selected={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should render door correctly with content prop and open", () => {
    const wrapper = shallow(<Door content={"content"} open={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
