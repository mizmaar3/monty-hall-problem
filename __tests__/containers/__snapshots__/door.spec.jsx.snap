// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Door container Should render door correctly with content prop and open 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#d4d4d4",
      "border": "1px solid #00f",
      "display": "flex",
      "height": "200px",
      "justifyContent": "center",
      "width": "125px",
    }
  }
>
  content
</div>
`;

exports[`Door container Should render door correctly with default props 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#999",
      "border": "",
      "display": "flex",
      "height": "200px",
      "justifyContent": "center",
      "width": "125px",
    }
  }
/>
`;

exports[`Door container Should render door correctly with open prop 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#d4d4d4",
      "border": "1px solid #00f",
      "display": "flex",
      "height": "200px",
      "justifyContent": "center",
      "width": "125px",
    }
  }
/>
`;

exports[`Door container Should render door correctly with selected prop 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "background": "#d4d4d4",
      "border": "4px solid #0f0",
      "display": "flex",
      "height": "200px",
      "justifyContent": "center",
      "width": "125px",
    }
  }
/>
`;
